File: Algorithm I/2. Two Pointers - Day 2, 3, 4, 5/P167.cpp
// Date: Mon, Sep 27, 2021
// 167. Two Sum II - Input array is sorted (https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/)

// time complexity: O(n)
// space complexity: O(1)
vector<int> twoSum(vector<int>& numbers, int target) {
	int left = 0, right = numbers.size() - 1;
	while (left < right) {
		int current = numbers[left] + numbers[right];
		if (current == target)
			return vector<int>{left + 1, right + 1};
		if (current > target)
			right--;
		else
			left++;
	}
	return vector<int>{-1, -1};
}

File: LeetCoding-Challenge/June-2022/P167.java
package leetcodingchallenge.june2022;

/**
 * Date: June 09, 2022
 * 167. Two Sum II - Input Array Is Sorted
 * https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
 */

public class P167 {
    // time: O(n)
    // space: O(1)
    public int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length - 1, sum;
        while (left < right) {
            sum = numbers[left] + numbers[right];
            if (sum > target) right--;
            else if (sum < target) left++;
            else break;
        }
        return new int[]{left + 1, right + 1};
    }
}
